lib/server/server.js

```diff
@@ -80,20 +96,29 @@ Meteor.methods({
         }
       });
     }
+    //console.log("docCheckThird",doc);
 
     if (doc) {
       // xxx: Hack
       // Clean up mech making sure tokens are uniq - android sometimes generate
       // new tokens resulting in duplicates
-      var removed = Push.appCollection.remove({
-        $and: [
-          { _id: { $ne: doc._id } },
-          { token: doc.token },     // Match token
-          { appName: doc.appName }, // Match appName
-          { token: { $exists: true } }  // Make sure token exists
-        ]
-      });
-
+      // var removed = Push.appCollection.remove({
+      //   $and: [
+      //     { _id: { $ne: doc._id } },
+      //     { token: doc.token },     // Match token
+      //     { appName: doc.appName }, // Match appName
+      //     { token: { $exists: true } }  // Make sure token exists
+      //   ]
+      // });
+      if(doc.token)
+      { 
+        var removed = Push.appCollection.remove({ 
+          $and: [ { _id: { $ne: doc._id } },
+          { token: doc.token }, // Match token 
+          { appName: doc.appName }, // Match appName 
+          { token: { $exists: true } } // Make sure token exists 
```

Comment:
No need for checking $exists as you are already checking in if(doc.token).

Reply:
I removed the commented out code